File: src/main/java/no/nav/bidrag/beregn/forholdsmessigfordeling/ForholdsmessigFordelingCore.java
package no.nav.bidrag.beregn.forholdsmessigfordeling;

import no.nav.bidrag.beregn.forholdsmessigfordeling.dto.BeregnForholdsmessigFordelingGrunnlagCore;
import no.nav.bidrag.beregn.forholdsmessigfordeling.dto.BeregnForholdsmessigFordelingResultatCore;
import no.nav.bidrag.beregn.forholdsmessigfordeling.periode.ForholdsmessigFordelingPeriode;


public interface ForholdsmessigFordelingCore {

  BeregnForholdsmessigFordelingResultatCore beregnForholdsmessigFordeling(
      BeregnForholdsmessigFordelingGrunnlagCore beregnForholdsmessigFordelingGrunnlagCore);

  static ForholdsmessigFordelingCore getInstance() {
    return new ForholdsmessigFordelingCoreImpl(ForholdsmessigFordelingPeriode.Companion.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/forholdsmessigfordeling/ForholdsmessigFordelingCore.kt
package no.nav.bidrag.beregn.forholdsmessigfordeling

import no.nav.bidrag.beregn.forholdsmessigfordeling.dto.BeregnForholdsmessigFordelingGrunnlagCore
import no.nav.bidrag.beregn.forholdsmessigfordeling.dto.BeregnForholdsmessigFordelingResultatCore
import no.nav.bidrag.beregn.forholdsmessigfordeling.periode.ForholdsmessigFordelingPeriode

fun interface ForholdsmessigFordelingCore {
    fun beregnForholdsmessigFordeling(grunnlag: BeregnForholdsmessigFordelingGrunnlagCore): BeregnForholdsmessigFordelingResultatCore

    companion object {
        fun getInstance(): ForholdsmessigFordelingCore = ForholdsmessigFordelingCoreImpl(ForholdsmessigFordelingPeriode.getInstance())
    }
}
